### ISSUE TITLE
no-unused-modules does not consider exported TypeScript interfaces, types and enums

### ISSUE THREAD

**Issue by felixfbecker** [benmosher/eslint-plugin-import#1680]
```ts
export interface Foo {}
```

with
```js
module.exports = {
  extends: [
	'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:import/typescript',
  ],
  rules: {
    'import/no-unused-modules': ['error', { missingExports: true }],
  }
}
```
produces `No exports found` error by `import/no-unused-modules`.

Expected:
No error.

**Comment by ljharb** [comment]
Seems like we'll have to handle files that only export types.

**Comment by boredland** [comment]
is there a workaround for files that only export interfaces?
